### PR TITLE
CONN-2451 Deferred Response Endpoint header

### PR DESCRIPTION
**Pull request by trantang**
Adding soap header back to the soap message

### CODE REVIEW

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/server/BaseService.java [hunk 1]
```diff
@@ -118,6 +119,12 @@ private void handleMessageId(AssertionType assertion, WebServiceContext context,
         }
     }
 
+    private void handleDeferredResponseEndpoint(AssertionType assertion, WebServiceContext context) {
+        if (NullChecker.isNullish(assertion.getDeferredResponseEndpoint())) {
```
Comment (TabassumJafri) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/server/BaseService.java]: Use StringUtils.isBlank here.

**Review by plobre — APPROVED** [review]

**Review by ptambellini01 — APPROVED** [review]

File: Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/auditrepository/nhinc/proxy/AuditRepositoryProxyWebServiceSecuredImpl.java [hunk 2]
```diff
@@ -68,18 +68,18 @@ public AcknowledgementType auditLog(LogEventRequestType request, AssertionType a
...
             }
         } catch (Exception e) {
             LOG.error("Failed to call the web service({}). An unexpected exception occurred. Exception: {}",
-                    NhincConstants.AUDIT_REPO_SECURE_SERVICE_NAME, e.getLocalizedMessage(), e);
+                NhincConstants.AUDIT_REPO_SECURE_SERVICE_NAME, e.getLocalizedMessage(), e);
         }
 
         LOG.debug("In AuditRepositoryProxyWebServiceSecured.auditLog(...) - completed called to ConnectionManager to "
-                + "retrieve endpoint.");
+            + "retrieve endpoint.");
```
Comment (mhpnguyen) [inline Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/auditrepository/nhinc/proxy/AuditRepositoryProxyWebServiceSecuredImpl.java]: Optional: use parameterized log if possible.

File: Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/auditrepository/nhinc/proxy/AuditRepositoryProxyWebServiceUnsecuredImpl.java [hunk 3]
```diff
@@ -66,18 +66,18 @@ public AcknowledgementType auditLog(LogEventRequestType request, AssertionType a
...
             }
         } catch (Exception e) {
             LOG.error("Failed to call the web service ({}).  An unexpected exception occurred. Exception: {}",
-                    NhincConstants.AUDIT_REPO_SERVICE_NAME, e.getLocalizedMessage(), e);
+                NhincConstants.AUDIT_REPO_SERVICE_NAME, e.getLocalizedMessage(), e);
         }
 
         LOG.debug("In AuditRepositoryProxyWebServiceUnsecuredImpl.auditLog(...) - completed called to "
-                + "ConnectionManager to retrieve endpoint.");
+            + "ConnectionManager to retrieve endpoint.");
```
Comment (mhpnguyen) [inline Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/auditrepository/nhinc/proxy/AuditRepositoryProxyWebServiceUnsecuredImpl.java]: Optional: use parameterized log if possible

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/nhinclib/NhincConstants.java [hunk 4]
```diff
@@ -354,6 +354,9 @@ public static EVENT_LOGGING_SERVICE_NAME fromValueString(String valueString) {
     public static final String WS_SOAP_HEADER_MESSAGE_ID = "MessageID";
     public static final String WS_SOAP_HEADER_MESSAGE_ID_PREFIX = "urn:uuid:";
     public static final String WS_SOAP_TARGET_HOME_COMMUNITY_ID = "TargetHomeCommunityID";
+    public static final String WS_IHE_DEFERRED_RESPONSE_ENDPOINT = "DeferredResponseEndpoint";
+    public static final String NS_IHE_ITI_2007 = "urn:ihe:iti:2007";
```
Comment (TabassumJafri) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/nhinclib/NhincConstants.java]: Can we put this in a properties file? The sample message says its TBD

File: Product/Production/Services/PolicyEngineCore/src/main/java/gov/hhs/fha/nhinc/policyengine/adapter/pep/proxy/AdapterPEPProxyWebServiceUnsecuredImpl.java [hunk 5]
```diff
@@ -86,13 +86,13 @@ public CheckPolicyResponseType checkPolicy(CheckPolicyRequestType request, Asser
...
                 checkPolicyResponse = (CheckPolicyResponseType) client.invokePort(AdapterPEPPortType.class,
-                        "checkPolicy", request);
+                    "checkPolicy", request);
             } else {
                 LOG.error("Failed to call the web service (" + serviceName + ").  The URL is null.");
             }
         } catch (Exception ex) {
             String message = "Error occurred calling AdapterPEPProxyWebServiceUnsecuredImpl.checkPolicy.  Error: "
-                    + ex.getMessage();
+                + ex.getMessage();
```
Comment (mhpnguyen) [inline Product/Production/Services/PolicyEngineCore/src/main/java/gov/hhs/fha/nhinc/policyengine/adapter/pep/proxy/AdapterPEPProxyWebServiceUnsecuredImpl.java:95]: Optional: no need to log if we rethrow exception (https://sonarcloud.io/organizations/default/rules?languages=java&open=squid%3AS2139&q=log)

**Review by mhpnguyen — APPROVED** [review]

**Review by TabassumJafri — CHANGES_REQUESTED** [review]

File: Product/Production/Services/PolicyEngineCore/src/main/java/gov/hhs/fha/nhinc/policyengine/adapter/pep/proxy/AdapterPEPProxyWebServiceUnsecuredImpl.java (hunk 5 above)
Comment (trantang) [reply Product/Production/Services/PolicyEngineCore/src/main/java/gov/hhs/fha/nhinc/policyengine/adapter/pep/proxy/AdapterPEPProxyWebServiceUnsecuredImpl.java:95]: deleted log.error

**Review by TabassumJafri — APPROVED** [review]

**Pull request merged**